spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigRefresh.java

```diff
@@ -74,31 +67,56 @@ public void setApplicationEventPublisher(ApplicationEventPublisher applicationEv
     /**
      * Checks configurations to see if they are no longer cached. If they are no longer
      * cached they are updated.
+     * 
+     * @return Future with a boolean of if a RefreshEvent was published. If
+     * refreshConfigurations is currently being run elsewhere this method will return
+     * right away as <b>false</b>.
      */
-    public void refreshConfigurations() {
-        if (this.running.compareAndSet(false, true)) {
-            Boolean refreshed = false;
-            Date notCachedTime = DateUtils.addSeconds(lastCheckedTime, Math.toIntExact(delay.getSeconds()));
-            Date date = new Date();
-            if (date.after(notCachedTime)) {
-                for (ConfigStore configStore : configStores) {
-                    String watchedKeyNames = watchedKeyNames(configStore, storeContextsMap);
-                    refreshed = refresh(configStore, CONFIGURATION_SUFFIX, watchedKeyNames);
-                    // Refresh Feature Flags
-                    if (!refreshed) {
-                        refreshed = refresh(configStore, FEATURE_SUFFIX, FEATURE_STORE_WATCH_KEY);
-                    }
+    public Future<Boolean> refreshConfigurations() {
+        return CompletableFuture.supplyAsync(() -> refreshStores());
+    }
 
-                    // The Refresh Event updates all config stores
-                    if (refreshed) {
-                        break;
+    /**
+     * Goes through each config store and checks if any of its keys need to be refreshed.
+     * If any store has a value that needs to be updated a refresh event is called after
+     * every store is checked.
+     * @return If a refresh event is called.
+     */
+    private boolean refreshStores() {
+        boolean needsRefresh = false;
```

Comment:
Rename it with a another word. In the java doc, it tells the user that If a refresh event is called, but needsRefresh makes users feel that this method will tell them if they need to do it or not.